src/panel/details.tsx

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/* eslint-disable indent */ // Allowing for some custom intent under svDetailsGrid 2D layout.
+
+import { autorun, computed, IObservableValue, observable } from 'mobx';
+import { observer } from 'mobx-react';
+import * as React from 'react';
+import { Component } from 'react';
+import ReactMarkdown from 'react-markdown';
+import { Result, ThreadFlowLocation } from 'sarif';
+import { parseArtifactLocation, parseLocation } from '../shared';
+import './details.scss';
+import { postSelectArtifact, postSelectLog } from './indexStore';
+import { List, renderMessageWithEmbeddedLinks, Tab, TabPanel } from './widgets';
+
+interface DetailsProps { result: Result, height: IObservableValue<number> }
+@observer export class Details extends Component<DetailsProps> {
+    private selectedTab = observable.box('Info')
+    @computed private get threadFlowLocations() {
+        return this.props.result?.codeFlows?.[0]?.threadFlows?.[0].locations
+            .filter(tfLocation => tfLocation.location);
+    }
+    constructor(props: DetailsProps) {
+        super(props);
+        autorun(() => {
+            const hasThreadFlows = !!this.threadFlowLocations?.length;
+            this.selectedTab.set(hasThreadFlows ? 'Code Flows' : 'Info');
```

Comment:
>'Code Flows' : 'Info' [](start = 50, length = 21)

Should strongly type the tabs.

Reply:
Added `type TabName = 'Info' | 'Code Flows'` and applied it.

---
In reply to: [441688737](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r441688737) [](ancestors = 441688737)